examples/BasicOperations/GetResponsiveSearchAds.php

```diff
@@ -0,0 +1,175 @@
+use Google\Ads\GoogleAds\V3\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V3\Services\GoogleAdsRow;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\Internal\RepeatedField;
+
+/**
+ * This example gets non-removed responsive search ads in a specified ad group.
+ * To add responsive search ads, run BasicOperations/AddResponsiveSearchAd.php.
+ * To get ad groups, run BasicOperations/GetAdGroups.php.
+ */
+class GetResponsiveSearchAds
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    // Optional: Specify an ad group ID below to restrict search to only a given ad group.
+    private const AD_GROUP_ID = null;
+
+    private const PAGE_SIZE = 1000;
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT,
+            ArgumentNames::AD_GROUP_ID => GetOpt::OPTIONAL_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID,
+                $options[ArgumentNames::AD_GROUP_ID] ?: self::AD_GROUP_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int|null $adGroupId the ad group ID for which responsive search ads will be retrieved.
+     *     If `null`, returns from all ad groups
```

Comment:
nit: "and by default"

Reply:
I'm not sure where to place your phrase? after "If `null`"?
Anyway, it should be already covered above in declaration of AD_GROUP_ID.
The parameter $adGroupId here doesn't specify = null, so I don't think technically saying "by default" here is 100% accurate.

As this is a nit, I'm skipping for now. If we feel we should add this wording, let's do it later (but I think the more appropriate place would be in the declaration of AD_GROUP_ID or at the file-level PHPDoc)

Reply:
There is no issue, I got confused by "?int" because I thought it mean "optional" but it does not.